pkg/velaql/view.go

```diff
@@ -131,6 +137,96 @@ func (handler *ViewHandler) delete(ctx context.Context, cluster string, owner co
 	return handler.cli.Delete(ctx, manifest)
 }
 
+// ValidateView makes sure the cue provided can use as view.
+//
+// For now, we only check 1. cue is valid 2. `status` field exists
+func ValidateView(viewStr string) error {
+	val, err := value.NewValue(viewStr, nil, "")
+	if err != nil {
+		return errors.Errorf("error when parsing view: %v", err)
+	}
+
+	// Make sure `status` field exists
+	v, err := val.LookupValue(DefaultExportValue)
+	if err != nil {
+		return errors.Errorf("no `status` field found in view: %v", err)
+	}
+	if _, err := v.String(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// ParseViewIntoConfigMap parses a CUE string (representing a view) into a ConfigMap
+// ready to be stored into etcd.
+func ParseViewIntoConfigMap(viewStr, name string) (*v1.ConfigMap, error) {
```

Comment:
This should become a reusable function. There is a `renderCUEView` function in pkg/addon/addon.go:L1003, they should merge.

Reply:
I merged this function into `renderCUEView`.